modules/javafx.graphics/src/main/java/javafx/application/Platform.java

```diff
@@ -302,6 +302,9 @@ public static boolean isSupported(ConditionalFeature feature) {
      * @throws IllegalStateException if this method is called on a thread
      * other than the JavaFX Application Thread.
      *
+     * @throws RuntimeException if this call would exceed the maximum
```

Comment:
I wonder if there is a more specific subclass of `RuntimeException` that we could throw? Maybe `IllegalStateException`, which seems the closest in meaning and has the advantage that we already throw it in other cases.

Reply:
I'll have to wait for others to chime in on that question, I don't have much background in this area.

Reply:
The `IllegalStateException` is the right one, as it states that it _"Signals that a method has been invoked at an illegal or inappropriate time."_

Another positive thing it won't require CSR as this method already throws one.